packages/block-library/src/heading/index.js

```diff
@@ -41,7 +36,7 @@ export const settings = {
 		}
 
 		if ( context === 'accessibility' ) {
-			return isEmpty( content )
+			return ! content || content.length === 0
```

Comment:
`content` will be a string, but we're still making sure it's not nullish for some weird reason.